tests/test_gdal_subsetter.py

```diff
@@ -11,63 +11,63 @@
+from gdal_subsetter.exceptions import DownloadError, UnsupportedFileFormatError
 from gdal_subsetter.transform import HarmonyAdapter
 
 
 class TestEndToEnd(TestCase):
     @classmethod
     def setUpClass(cls):
-        """ Set test fixtures that can be shared between tests. """
+        """Set test fixtures that can be shared between tests."""
         cls.config = harmony_config(validate=False)
 
     def setUp(self):
-        """ Create test fixtures that must be unique to each test. """
+        """Create test fixtures that must be unique to each test."""
         self.temp_dir = mkdtemp()
 
     def tearDown(self):
-        """ Remove test fixtures that are unique to each test. """
+        """Remove test fixtures that are unique to each test."""
         if exists(self.temp_dir):
             rmtree(self.temp_dir)
 
     @patch('gdal_subsetter.transform.download')
     def test_failed_download(self, mock_download):
-        """ Ensure a failed download raises a service exception, and that the
-            output error message is informative.
-
-        """
-        exception_message = 'Download exception message'
-        mock_download.side_effect = Exception(exception_message)
+        """Ensure a failed download raises a service exception."""
+        raw_exception_message = 'Download exception message'
+        mock_download.side_effect = Exception(raw_exception_message)
 
         stac_catalog = Catalog('input catalog', 'description')
         stac_item = Item('input', {}, [0, 0, 1, 1], datetime(2020, 1, 1), {})
         stac_item.add_asset('data', Asset('url.com/file.nc4', roles=['data']))
         stac_catalog.add_item(stac_item)
 
-        harmony_message = Message({'accessToken': 'fake-token',
-                                   'callback': 'https://example.com',
-                                   'format': {'mime': 'image/png'},
-                                   'sources': [{'collection': 'C1234-XYZ'}],
-                                   'stagingLocation': 's3://example-bucket',
-                                   'user': 'ascientist'})
-
-        adapter = HarmonyAdapter(harmony_message, catalog=stac_catalog,
-                                 config=self.config)
+        harmony_message = Message(
+            {
+                'accessToken': 'fake-token',
+                'callback': 'https://example.com',
+                'format': {'mime': 'image/png'},
+                'sources': [{'collection': 'C1234-XYZ'}],
+                'stagingLocation': 's3://example-bucket',
+                'user': 'ascientist',
+            }
+        )
+
+        adapter = HarmonyAdapter(
+            harmony_message, catalog=stac_catalog, config=self.config
+        )
 
         with self.assertRaises(DownloadError) as context_manager:
             adapter.invoke()
 
-            self.assertTrue(
-                str(context_manager.exception).endswith(exception_message)
-            )
```

Comment:
Feel a bit daft - having this assertion in the `context_manager` stops it being actually executed. Schoolboy error on my part.

I was wary of updating `harmony-service-lib-py` to 2.8.0, because that's a major version bump. Feels like something to do separately and not rock the boat while doing other things.